File: src/easy/TwoSum.js
var twoSum = function(nums, target) {
  var diffMap = {};
  for (var i = 0; i < nums.length; i++) {
    var cur = nums[i];
    var diff = target - cur;
    if (diffMap[diff] !== undefined) {
      return [diffMap[diff], i];
    }
    diffMap[cur] = i;
  }
};

File: src/easy/TwoSum.cpp
class Solution {
public:
  vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> res;
    map<int, int> map;
    for (int i = 0; i < nums.size(); i++) {
      int cur = nums[i];
      int diff = target - cur;
      if (map.count(diff) != 0) {
        res.push_back(map[diff]);
        res.push_back(i);
        return res;
      }
      map[cur] = i;
    }
  }
};